From e48a83129f386bbb4c1cfdae23f987b2397c7e58 Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <b37945@freescale.com>
Date: Thu, 11 Apr 2013 15:22:41 +0800
Subject: [PATCH 448/635] ENGR00215174 fix stop preview may cause process
 crash.

when preview stream stop called, it will remove frame listerner.
but, device thread may call dispatchFrame and use null pointer.
so, stop device thread firstly, then call preview stream to stop.

Signed-off-by: Xiaowen Liu <b37945@freescale.com>
---
 mx6/libcamera2/PreviewStream.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mx6/libcamera2/PreviewStream.cpp b/mx6/libcamera2/PreviewStream.cpp
index 77b1a78..7ab0e11 100755
--- a/mx6/libcamera2/PreviewStream.cpp
+++ b/mx6/libcamera2/PreviewStream.cpp
@@ -195,11 +195,11 @@ int PreviewStream::start()
 int PreviewStream::stop()
 {
     FLOG_TRACE("PreviewStream %s running", __FUNCTION__);
-    StreamAdapter::stop();
-
     if (mDeviceAdapter.get() != NULL) {
         mDeviceAdapter->stopPreview();
     }
+
+    StreamAdapter::stop();
     return NO_ERROR;
 }
 
-- 
1.8.0

